Borrowed from the curlftpfs_0.9.2-5 Debian package.

Fix a memory leak. See Bug #587250.
--- a/ftpfs.c	2008-04-30 01:05:47.000000000 +0200
+++ a/ftpfs.c.slesimple	2010-01-01 22:12:10.000000000 +0100
@@ -615,6 +615,8 @@ static void free_ftpfs_file(struct ftpfs
   sem_destroy(&fh->data_need);
   sem_destroy(&fh->data_written);
   sem_destroy(&fh->ready);
+  if (fh->buf.size) { buf_free(&fh->buf); }
+  if (fh->stream_buf.size) { buf_free(&fh->stream_buf); }
   free(fh);
 }
 
